File: golang/monothread_sort/selection_sort.go
package monothread_sort

type selectionSort struct{}

func NewSelectionSort() *selectionSort {
	return &selectionSort{}
}

func (ss *selectionSort) Sort(array []int) {
	for i := 0; i < len(array); i++ {
		index := i

		for j := i + 1; j < len(array); j++ {
			if array[j] < array[index] {
				index = j
			}
		}

		if index != i {
			array[i], array[index] = array[index], array[i]
		}
	}
}

File: java/monothread_sort/SelectionSort.java
package monothread_sort;

/**
 * SelectionSort
 */
public class SelectionSort {
    public void sort(int[] array) {
        for (int i = 0; i < array.length; i++) {
            int index = i;

            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[index]) {
                    index = j;
                }
            }

            if (index != i) {
                int aux = array[i];

                array[i] = array[index];
                array[index] = aux;
            }
        }
    }
}
